### ISSUE TITLE
Error when excluding only addons or folders

### ISSUE THREAD

**Issue by DennisOtt-git-hub** [jcwillox/hass-auto-backup#43]
I recently upgraded to 2021.8.7 and started getting snapshot failures when I run this integration in a script.  The log message follows:

Logger: homeassistant.components.websocket_api.http.connection
Source: custom_components/auto_backup/__init__.py:250 
Integration: Home Assistant WebSocket API (documentation, issues) 
First occurred: 11:14:42 AM (1 occurrences) 
Last logged: 11:14:42 AM

[2931029472] Error handling message: Unknown error
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 25, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 525, in handle_execute_script
    await script_obj.async_run(msg.get("variables"), context=context)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1219, in async_run
    await asyncio.shield(run.async_run())
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 353, in async_run
    await self._async_step(log_exceptions=False)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 371, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 574, in _async_call_service_step
    await self._async_run_long_action(service_task)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 535, in _async_run_long_action
    long_task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1491, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1526, in _execute_service
    await handler.job.target(service_call)
  File "/usr/src/homeassistant/homeassistant/components/script/__init__.py", line 278, in service_handler
    await script_entity.async_turn_on(
  File "/usr/src/homeassistant/homeassistant/components/script/__init__.py", line 382, in async_turn_on
    await coro
  File "/usr/src/homeassistant/homeassistant/components/script/__init__.py", line 409, in _async_run
    return await self.script.async_run(script_vars, context)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1219, in async_run
    await asyncio.shield(run.async_run())
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 353, in async_run
    await self._async_step(log_exceptions=False)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 371, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 571, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1491, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1526, in _execute_service
    await handler.job.target(service_call)
  File "/config/custom_components/auto_backup/__init__.py", line 154, in snapshot_service_handler
    await auto_backup.async_create_backup(data)
  File "/config/custom_components/auto_backup/__init__.py", line 307, in async_create_backup
    addons, folders = self.ensure_slugs(include or exclude, installed_addons)
  File "/config/custom_components/auto_backup/__init__.py", line 250, in ensure_slugs
    addons = inclusion[ATTR_ADDONS]
KeyError: 'addons'

I would appreciate some assistance with this...

**Comment by jcwillox** [comment]
Ahh damn, nice catch, this is 100% a bug in my code. I rewrote a large portion of the codebase in version `0.8.0` almost without introducing any new bugs. This also appears to be the only case I didn't test 'excluding only addons or folders'.

I'll release a fixed version tonight, for the time being you can just set addons or folders to an empty array
```yaml
service: auto_backup.snapshot_full
data:
  name: Backup Exclude Test
  exclude:
    addons: []
    folders:
      - share
      - local add-ons
```

**Comment by jcwillox** [comment]
This issue should be fixed in version `0.8.1` (for HA <2021.9) and in `0.9.2`, I've also updated the release asset for `0.8.0` to include the fix.

Unfortunately, you may have trouble updating to `0.8.1` if you are currently running HA 2021.8 or any version prior HA 2021.9, see this issue for more information https://github.com/jcwillox/hass-auto-backup/issues/40#issuecomment-907588919, but basically, HACS won't let you update and you'll have to manually update it, or alternatively update to HA 2021.9 (either beta or wait for the release).
